src/interfaces/IUpgradeRegressionManager.sol

```diff
@@ -1,9 +1,27 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.30;
 
+// External Libraries
+import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
+
 // Internal Libraries
 import {ITimelockTarget} from "interfaces/ITimelockTarget.sol";
 
+/// @notice Struct to store rollback data.
+/// @param queueExpiresAt The timestamp before which the rollback must be queued for execution.
+/// @param executableAt The timestamp after which the rollback can be executed.
+/// @param executed Whether the rollback has been executed.
+/// @param canceled Whether the rollback has been canceled.
+/// @dev executed and canceled are mutually exclusive - both cannot be true
+///      queueExpiresAt must be > block.timestamp when rollback is proposed
+///      executableAt must be > block.timestamp when rollback is queued
+struct Rollback {
+  uint256 queueExpiresAt;
+  uint256 executableAt;
```

Comment:
This can/probably should be pulled into an issue and done in a separate PR, but it's probably worth changing these to uint48 types, which is what the Governor uses internally for timestamps. Then the whole struct will fit into 1 storage slot, which will save on gas within various operations (not that we're too concerned with gas golfing, but it's an easy change)

Reply:
This is simple enough change that it can be done in this pr with the safecast library
https://github.com/ScopeLift/governance-rollback/pull/29/commits/fcc499b2c3ec3e2a373e1fd4b1f79775ff8d3b2b